pubsub/pubsub.go

```diff
@@ -16,6 +16,7 @@ type Message interface {
 // A Publisher takes some data and publishes it on a topic
 type Publisher interface {
 	Publish(context.Context, []byte) error
+	PublishUntilComplete(context.Context, []byte) error
```

Comment:
it might be best to make a separate interface so we don't break existing code if it needs to upgrade